src/System.CommandLine/Binding/Binder.cs

```diff
@@ -7,15 +7,50 @@ namespace System.CommandLine.Binding
 {
     internal static class Binder
     {
-        internal static bool IsMatch(this string parameterName, string alias) =>
-            string.Equals(alias?.RemovePrefix()
-                              .Replace("-", "")
-                              .Replace("|", "or"),
-                          parameterName,
-                          StringComparison.OrdinalIgnoreCase);
+        internal static bool IsMatch(this string parameterName, string alias)
+        {
+            if (alias is null || parameterName is null)
+            {
+                return false;
+            }
+
+            var parameterIndex = 0;
+            for (var aliasIndex = alias.AdvancePrefix(); aliasIndex < alias.Length; aliasIndex++)
+            {
+                var aliasChar = alias[aliasIndex];
+
+                if (aliasChar == '-')
+                {
+                    continue;
+                }
+
+                var parameterNameChar = parameterName[parameterIndex];
+
+                if (aliasChar == '|')
+                {
+                    if ((parameterName.Length < parameterIndex + 2) || ((parameterNameChar | 32) != 'o') || ((parameterName[parameterIndex + 1] | 32) != 'r'))
```

Comment:
If you set that bit (00100000) on a character and it becomes an **o** or **r**, it's because it was, respectively, an **O** or **R**. It's a no-op for **o** or **r** and it doesn't matter for other characters.

Reply:
Out of curiosity, what's the performance implication of this versus a `char.Equals` or `string.Equals`?

One aspect of this particular feature that's worth calling out is that it's a localization problem, so the implementation is going to need to change to take translations into account at some point. That might disrupt the optimization, and at the very least it makes the logic here a little harder to follow.

Reply:
It's not about `char.Equals` or `string.Equals`.

The original code was:

```C#
internal static bool IsMatch(this string parameterName, string alias) =>
        string.Equals(alias?.RemovePrefix()
                          .Replace("-", "")
                          .Replace("|", "or"),
                      parameterName,
                      StringComparison.OrdinalIgnoreCase);
```

It's creating 3 extra strings for an `OrdinalIgnoreCase` comparison.

Every time there's an avoidable string creation, a bucket of CO2 is thrown into the atmosphere. 😄 

Now that we're streaming the string, there are more efficient ways to compare ordinal characters.

The original code is not particularly localizable. By the time we get to that, there might be a code generator for regular expressions that may allow for a better localization.

But, the truth is, I hadn't thought about that.